### PR TITLE
improve blurry image placeholder generation

### PR DESCRIPTION
**Pull request by sebastianbenz**
* remove restrictions for `noloading` attribute. I realised that with
blurred placeholder images in particular you don't want to have a
loading indicator.
* add support for image URLs using query and hash parameters
* add support for fill and intrinsic layouts which both are also used in
responsive contexts.
* make number of blurred images configurable via
`maxBlurredPlaceholders` option.

### CODE REVIEW

**Comment by choumx** [comment]
Thanks for improving this. 

/cc @jridgewell FYI

File: packages/optimizer/lib/transformers/AddBlurryImagePlaceholders.js [hunk 1]
```diff
@@ -183,8 +186,10 @@ class AddBlurryImagePlaceholders {
    */
   resolvePath_(base, path) {
     try {
-      return new URL(path, base).toString();
+      return new URL(path).toString();
     } catch (e) {
+      // not an absolute URL
+      path = new URL(path, 'https://example.com').pathname;
```
[the remark this answers is not in the document]
Comment (sebastianbenz) [reply packages/optimizer/lib/transformers/AddBlurryImagePlaceholders.js]: Removing the query string is intended to be able to resolve images on the file system. However, there's one case not yet supported, which is when base specifies an image host. Will fix this.

**Comment by sebastianbenz** [comment]
@jridgewell that's a matter of definition. The current behaviour is that images show a loading indicator and a blurry image placeholder, which IMO is usually not the desired behaviour. Hence my change, to make it possible to remove the loading indicator while still showing a blurry image placeholder. 

But you're right. `noloading` should always disable any loading indicator (including blurry image). Otherwise, the loading indication would potentially not be consistent across cache and origin (with SSR). A better approach would be to always disable the loading indicator for blurry image placeholders. WDYT?

**Comment by jridgewell** [comment]
> The current behaviour is that images show a loading indicator and a blurry image placeholder, which IMO is usually not the desired behaviour. Hence my change, to make it possible to remove the loading indicator while still showing a blurry image placeholder.

I think this needs to be changed in Runtime, not in blurry image creation.

File: packages/optimizer/lib/transformers/AddBlurryImagePlaceholders.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (sebastianbenz) [reply packages/optimizer/lib/transformers/AddBlurryImagePlaceholders.js]: done

**Comment by sebastianbenz** [comment]
Adressed comments:

* Support resolving absolute and relative URLs
* Don't show a blurry image placeholder if the noloading attribute is
present
* Add noloading attribute when adding a blurry image placeholder

@jridgewell: I've fixed it for now by adding the `noloading` attribute in the transformer. Not sure if there's a better way to do this on runtime side. Demo here: https://amp-demos.glitch.me/blurry-demo.html

PTAL

**Comment by sebastianbenz** [comment]
Thanks @jridgewell and @choumx!

**Pull request merged**